reconcile/external_resources/state.py

```diff
@@ -38,6 +38,15 @@ class ResourceStatus(StrEnum):
     PENDING_SECRET_SYNC: str = "PENDING_SECRET_SYNC"
     RECONCILIATION_REQUESTED: str = "RECONCILIATION_REQUESTED"
 
+    def is_in_progress(self) -> bool:
+        return self in {ResourceStatus.IN_PROGRESS, ResourceStatus.DELETE_IN_PROGRESS}
+
+    def needs_secret_sync(self) -> bool:
+        return self == ResourceStatus.PENDING_SECRET_SYNC
+
+    def has_errors(self) -> bool:
+        return self == ResourceStatus.ERROR
+
```

Comment:
Please change all three methods to be properties.

Reply:
done